pkg/comp-functions/functions/vshnkeycloak/deploy.go

```diff
@@ -583,3 +614,40 @@ exit 0`,
 	}
 	return extraInitContainersMap, nil
 }
+
+func copyConfig(comp *vshnv1.VSHNKeycloak, svc *runtime.ServiceRuntime) error {
+
+	cmObjectName := comp.GetName() + "-config-claim-observer"
+
+	cmClaimObserver := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      *comp.Spec.Parameters.Service.Configuration,
+			Namespace: comp.GetClaimNamespace(),
+		},
+	}
+	err := svc.SetDesiredKubeObserveObject(cmClaimObserver, cmObjectName)
+
+	if err != nil {
+		svc.Log.Info("Can't observe Keycloak config Configmap")
+		svc.AddResult(runtime.NewFatalResult(fmt.Errorf("cannot get configMap: %w", err)))
+	}
+
+	configMapClaim := &corev1.ConfigMap{}
+	err = svc.GetObservedKubeObject(configMapClaim, cmObjectName)
+	if err != nil {
+		svc.Log.Info("cannot get observed Keycloak config map")
+		svc.AddResult(runtime.NewWarningResult("cannot get observed Keycloak config map"))
+	}
+
+	configMapInstance := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        *comp.Spec.Parameters.Service.Configuration,
+			Namespace:   comp.GetInstanceNamespace(),
+			Labels:      configMapClaim.Labels,
+			Annotations: configMapClaim.Annotations,
+		},
+		Data: configMapClaim.Data,
+	}
+
+	return svc.SetDesiredKubeObject(configMapInstance, comp.GetName()+"config-map")
```

Comment:
My only concern here is as discussed in a meeting:
I'd create a follow up to engineer restart mechanism if customer update the cofigmap. Otherwise changes would be applied on next maintenance, or after we do manual restart after ticket.